WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -48,6 +51,9 @@ def create
         @competition.errors[:competition_id_to_clone] = "invalid"
       end
     end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates << current_user
+    end
 
     if @competition.errors.size == 0 && @competition.save
```

Comment:
This is "if we haven't run into any problems *yet*, lets try to save. See line 51. Perhaps it's possible to move this cloning logic down into the model more?